### DISCUSSION TITLE
How to use Moes BRT-100-TRV programming mode

### DISCUSSION THREAD

**Discussion by tom-ch1** [Koenkk/zigbee2mqtt#9430] [Devices]
Hi, I have a Moes BRT-100-TRV and integrated it into openHAB. Everything works fine so far, except for one thing:
I do not understand how to set a weekly program. The documentation is somewhat cryptic:
https://www.zigbee2mqtt.io/devices/BRT-100-TRV.html#programming_mode-composite
Can anybody supply an example, like e.g. the one for TuYa TS0601:
https://www.zigbee2mqtt.io/devices/TS0601_thermostat.html#controlling-device-specific-features

Thanks
Tom

**Comment by DodgerArt** [comment]
After manual setup in Zigbee2MQTT I see something like that

program_weekday {"weekday":" 6h:0m 21°C, 11h:30m 21°C, 13h:30m 23°C, 17h:30m 23°C "}

**Reply by tom-ch1** [reply]
Yes, but that does not answer my question :-)
I can confirm that (according to https://github.com/Koenkk/zigbee2mqtt/issues/9530#issuecomment-972823977) you can set the weekly schedule by publishing the following payload to `zigbee2mqtt/FRIENDLY_NAME/set`:
`{
  "programming_mode": "<w_1>  <w_2>  <w_3>  <w_4>  <sat_1>  <sat_2>  <sat_3>  <sat_4>  <sun_1>  <sun_2>  <sun_3>  <sun_4>"
}`
where `<w_n>` ist the nth schedule for weekdays, `<sat_n>` for saturdays, `<sun_n>` for sundays.
The schedules take the form `hh:mm/dd°C` where `hh` is the hour, `mm` the minute, and `dd` the temperature.
This is not documented anywhere and I'll try to get this into the documentation. I assume that you must specify exactly 4 definitions for weekdays, saturdays and sundays but I am not certain.

example schedule payload:
`{"programming_mode": "00:00/16°C  17:00/20°C  23:59/16°C  23:59/16°C  00:00/16°C  09:00/20°C  23:59/16°C  23:59/16°C  00:00/16°C  09:00/20°C  23:59/16°C  23:59/16°C"}'`
